### PR TITLE
feature: support next/config runtime configuration

### PR DESCRIPTION
**Pull request by toomuchdesign**
## What kind of change does this PR introduce?

Feature

## What is the current behaviour?

`next/config` is not set and always returns `undefined`.

## What is the new behaviour?

Support [Next.js runtime configuration](https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration) and set different `next/config` values based on `server` or `client` env. 

## Does this PR introduce a breaking change?

Yes.

## Please check if the PR fulfills these requirements:

- [X] Tests for the changes have been added
- [x] Docs have been added / updated

### CODE REVIEW

File: src/setNextRuntimeConfig.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import * as envConfig from 'next/dist/next-server/lib/runtime-config';
```
Comment (Meemaw) [inline src/setNextRuntimeConfig.ts:1]:
```suggestion
import { setConfig } from 'next/dist/next-server/lib/runtime-config';
```

File: src/setNextRuntimeConfig.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+export default function setNextRuntimeConfig({
+  runtimeEnv,
+}: {
+  runtimeEnv: 'server' | 'client';
+}): void {
+  const config = getNextConfig();
+  const { serverRuntimeConfig, publicRuntimeConfig } = config;
+
+  envConfig.setConfig({
```
Comment (Meemaw) [inline src/setNextRuntimeConfig.ts:12]:
```suggestion
  setConfig({
```

File: src/setNextRuntimeConfig.ts (hunk 1 above)
Comment (Meemaw) [inline src/setNextRuntimeConfig.ts:1]: Would it make more sense to keep this function along with the others in one place? (`src/nextConfig`).

**Review by Meemaw — APPROVED** [review]
Looks good. Left some minor comments.

File: src/setNextRuntimeConfig.ts (hunk 1 above)
Comment (toomuchdesign) [reply src/setNextRuntimeConfig.ts]: My take here (I could have misunderstood Next.js), is that `next.config.js` and Next.js runtime configuration (`next/config`) are 2 unrelated moving parts of Next.js.

`next/config` depends on `next.config.js` as much as many other Next.js parts do. On our case we also use `next.config.js` to get the allowed file extensions.

But maybe they are closer than I though and keeping them together might make sense. How do you see it?

Comment (Meemaw) [reply src/setNextRuntimeConfig.ts]: Agree those are 2 different parts. LGTM

**Pull request merged**